src/contracts/QBridge.h

```diff
@@ -0,0 +1,362 @@
+#pragma once
+
+#include "../core/src/contracts/qpi.h"
+
+using namespace QPI;
+
+
+struct BridgeContract : public ContractBase {
+
+public:
+    // Bridge Order Structure
+    struct BridgeOrder {
+        uint64 orderId;                      // Unique ID for the order
+        id qubicSender;                      // Sender address on Qubic
+        id ethAddress;                       // Destination Ethereum address
+        uint64 amount;                       // Amount to transfer
+        uint8 orderType;                     // Type of order (e.g., mint, transfer)
+        uint8 status;                        // Order status (e.g., Created, Pending, Refunded)
+        bit fromQubicToEthereum;             // Direction of transfer
+    };
+
+    // Input and Output Structs
+    struct createOrder_input {
+        id ethAddress;
+        uint64 amount;
+        bit fromQubicToEthereum;
+    };
+
+    struct createOrder_output {
+        uint8 status;
+        array<uint8, 32> message;
+    };
+
+    struct completeOrder_input {
+        uint64 orderId;
+    };
+
+    struct completeOrder_output {
+        uint8 status;
+        array<uint8, 32> message;
+    };
+
+    struct refundOrder_input {
+        uint64 orderId;
+    };
+
+    struct refundOrder_output {
+        uint8 status;
+        array<uint8, 32> message;
+    };
+
+    // Order Response Structure 
+    struct OrderResponse {
+        uint64 orderId;                      // Order ID as uint64
+        id originAccount;                    // Origin account
+        id destinationAccount;               // Destination account
+        uint64 amount;                       // Amount as uint64
+        array<uint8, 64> memo;               // Notes or metadata
+        uint32 sourceChain;                  // Source chain identifier
+    };
+
+    struct getOrder_input {
+        uint64 orderId;
+    };
+
+    struct getOrder_output {
+        uint8 status;
+        array<uint8, 32> message;
+        OrderResponse order;                 // Updated response format
+    };
+
+
+private:
+    // Contract State
+    QPI::HashMap<uint64, BridgeOrder, 256> orders; // Storage for orders (fixed size)
+    uint64 nextOrderId;                            // Counter for order IDs
+    uint64 lockedTokens;                           // Total locked tokens in the contract (balance)
+    uint64 transactionFee;                         // Fee for creating an order
+    id admin;                                      // Admin address
+    QPI::HashMap<id, bit, 16> managers;            // Managers list
+    uint64 totalReceivedTokens;                    // Total tokens received
+
+    // Internal methods for admin/manager permissions using qpi CALLS
+    PRIVATE_FUNCTION_WITH_LOCALS(isAdmin)
```

Comment:
'll refactor isAdmin and isManager to use PRIVATE_FUNCTION instead of PRIVATE_FUNCTION_WITH_LOCALS. I'll also add the necessary typedefs for input and output in both cases